src/Splunk.ModularInputs/Splunk/ModularInputs/Event.cs

```diff
@@ -69,7 +69,7 @@ public void ToXml(XmlWriter writer)
             {
                 double timestamp = (Time.Value.Ticks - ticksSinceEpoch) / TimeSpan.TicksPerSecond;
                 writer.WriteStartElement("time");
-                writer.WriteString(timestamp.ToString());
+                writer.WriteString(string.Format("{0}.{1:D3}", timestamp, Time.Value.Millisecond));
                 writer.WriteEndElement();
             }
```

Comment:
Thanks for this pull request. At present we are losing precision because we're doing integer, not floating point division. I would prefer this change which implements floating point division:
    double timestamp = (double)(Time.Value.Ticks - ticksSinceEpoch) / TimeSpan.TicksPerSecond;
    writer.WriteStartElement("time");
    writer.WriteString(timestamp.ToString());
We will then get the full precision of the timestamp based on the value of Time.Value.Ticks and the accuracy of the system clock.